### ISSUE TITLE
RLC-410 truncates password at 31 characters

### ISSUE THREAD

**Issue by charlesmunger** [fwestenberg/reolink_dev#104]
**Describe the bug**
Reolink cameras truncate passwords at 31 characters silently - and they do it in the login page, so if you pass the full 32-character password that you set in the camera's settings to this amazing integration, it will fail. This far from a security best practice, but I think the HA integration should probably behave the same as the camera's own login page.

**To Reproduce**
1. Create an account on your reolink camera, with a 32 character password.
2. Log in to the account via webUI, observe that it works
3. Use the same password to set up the reolink integration in HomeAssistant.

I figured out that they were truncating by going to the camera's login page, opening the network tab of chrome developer tools, and seeing that the password that actually gets sent out in the login request is truncated to 31 characters. Specifying the 31-character truncated version of the password when setting up the HA integration works. I think you just need a `[:31]` when sending out the password.

**Expected behavior**
Camera should be added successfully. In reality, we get an error.

**Environment:**
 - Home Assistant version: HassOS 4.17, 2020.12.6
 - Reolink camera model: RLC-410-5MP
 - Camera firmware number: 
```
Build No. build 20121100
Hardware No. IPC_51516M5M
Config Version v3.0.0.0
Firmware Version v3.0.0.136_20121100
```

**Additional context**
```
2020-12-15 21:13:17 DEBUG (MainThread) [custom_components.reolink_dev.base] {'port': 80, 'host': '10.0.0.133', 'username': 'REDACTED', 'password': '32 character password goes here!'}
2020-12-15 21:13:17 DEBUG (MainThread) [custom_components.reolink_dev.base] []
2020-12-15 21:13:17 DEBUG (MainThread) [reolink.camera_api] Reolink camera with host 10.0.0.133:80 trying to login with user REDACTED
2020-12-15 21:13:18 DEBUG (MainThread) [reolink.camera_api] Get response from 10.0.0.133: [{'cmd': 'Login', 'code': 1, 'error': {'detail': 'param error', 'rspCode': -4}}]
2020-12-15 21:13:18 ERROR (MainThread) [reolink.camera_api] Failed to login at IP x.x.x.x. Connection error.
2020-12-15 21:13:18 ERROR (MainThread) [reolink.camera_api] Error translating Reolink settings response
```

**Comment by fwestenberg** [comment]
Thanks! It's clear to me and I will add it to the next release.

**Comment by fwestenberg** [comment]
Can you check if it works with the latest version? Perhaps it should also be added to the motion event subscription. Let me know!

**Comment by charlesmunger** [comment]
Yes, tested the new version and it works. Thanks!

fwestenberg/reolink@7089c45677ccfbde26309a682215e9f7903c1c7e

**Comment by fwestenberg** [comment]
> Yes, tested the new version and it works. Thanks!
> 
> [fwestenberg/reolink@7089c45](https://github.com/fwestenberg/reolink/commit/7089c45677ccfbde26309a682215e9f7903c1c7e)

Motion detection also working?

**Comment by charlesmunger** [comment]
Motion detection does not appear to be working, shows as "Unavailable" in Lovelace. I see 

```
2020-12-26 22:55:47 ERROR (MainThread) [custom_components.reolink_dev.base] Host 10.0.0.133 error renewing the Reolink subscription
2020-12-26 22:55:47 DEBUG (MainThread) [reolink.subscription_manager] Reolink host 10.0.0.133 got response status: 400. Payload: {<?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://www.w3.org/2003/05/soap-envelope" xmlns:SOAP-ENC="http://www.w3.org/2003/05/soap-encoding" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:wsa="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:wsdd="http://schemas.xmlsoap.org/ws/2005/04/discovery" xmlns:wsa5="http://www.w3.org/2005/08/addressing" xmlns:xmime="http://tempuri.org/xmime.xsd" xmlns:xmime5="http://www.w3.org/2005/05/xmlmime" xmlns:xop="http://www.w3.org/2004/08/xop/include" xmlns:wsrfbf="http://docs.oasis-open.org/wsrf/bf-2" xmlns:tt="http://www.onvif.org/ver10/schema" xmlns:wstop="http://docs.oasis-open.org/wsn/t-1" xmlns:wsrfr="http://docs.oasis-open.org/wsrf/r-2" xmlns:ns1="http://www.onvif.org/ver10/actionengine/wsdl" xmlns:tev="http://www.onvif.org/ver10/events/wsdl" xmlns:ns10="http://www.onvif.org/ver10/events/wsdl/PullPointBinding" xmlns:ns11="http://www.onvif.org/ver10/events/wsdl/CreatePullPointBinding" xmlns:ns12="http://www.onvif.org/ver10/events/wsdl/PausableSubscriptionManagerBinding" xmlns:ns13="http://www.onvif.org/ver10/network/wsdl/RemoteDiscoveryBinding" xmlns:ns14="http://www.onvif.org/ver10/network/wsdl/DiscoveryLookupBinding" xmlns:tdn="http://www.onvif.org/ver10/network/wsdl" xmlns:ns3="http://www.onvif.org/ver20/analytics/wsdl/RuleEngineBinding" xmlns:ns4="http://www.onvif.org/ver20/analytics/wsdl/AnalyticsEngineBinding" xmlns:tan="http://www.onvif.org/ver20/analytics/wsdl" xmlns:ns5="http://www.onvif.org/ver10/events/wsdl/PullPointSubscriptionBinding" xmlns:ns6="http://www.onvif.org/ver10/events/wsdl/EventBinding" xmlns:ns7="http://www.onvif.org/ver10/events/wsdl/SubscriptionManagerBinding" xmlns:ns8="http://www.onvif.org/ver10/events/wsdl/NotificationProducerBinding" xmlns:wsnt="http://docs.oasis-open.org/wsn/b-2" xmlns:ns9="http://www.onvif.org/ver10/events/wsdl/NotificationConsumerBinding" xmlns:tad="http://www.onvif.org/ver10/analyticsdevice/wsdl" xmlns:tds="http://www.onvif.org/ver10/device/wsdl" xmlns:timg="http://www.onvif.org/ver20/imaging/wsdl" xmlns:tls="http://www.onvif.org/ver10/display/wsdl" xmlns:tmd="http://www.onvif.org/ver10/deviceIO/wsdl" xmlns:tptz="http://www.onvif.org/ver20/ptz/wsdl" xmlns:trc="http://www.onvif.org/ver10/recording/wsdl" xmlns:trp="http://www.onvif.org/ver10/replay/wsdl" xmlns:trt="http://www.onvif.org/ver10/media/wsdl" xmlns:trv="http://www.onvif.org/ver10/receiver/wsdl" xmlns:tse="http://www.onvif.org/ver10/search/wsdl" xmlns:ter="http://www.onvif.org/ver10/error" xmlns:tns1="http://www.onvif.org/ver10/topics" xmlns:wsse="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd" xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd"><SOAP-ENV:Header></SOAP-ENV:Header><SOAP-ENV:Body><SOAP-ENV:Fault SOAP-ENV:encodingStyle="http://www.w3.org/2003/05/soap-encoding"><SOAP-ENV:Code><SOAP-ENV:Value>SOAP-ENV:Sender</SOAP-ENV:Value><SOAP-ENV:Subcode><SOAP-ENV:Value>ter:NotAuthorized</SOAP-ENV:Value></SOAP-ENV:Subcode></SOAP-ENV:Code><SOAP-ENV:Reason><SOAP-ENV:Text xml:lang="en">Sender not Authorized</SOAP-ENV:Text></SOAP-ENV:Reason></SOAP-ENV:Fault></SOAP-ENV:Body></SOAP-ENV:Envelope>
}
2020-12-26 22:55:47 DEBUG (MainThread) [custom_components.reolink_dev] Finished fetching reolink data in 0.588 seconds```

In logs. 

I also encountered

```
Logger: homeassistant.config_entries
Source: custom_components/reolink_dev/__init__.py:176
First occurred: 10:53:20 PM (1 occurrences)
Last logged: 10:53:20 PM

Error unloading entry Camera1 for reolink_dev
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 321, in async_unload
    result = await component.async_unload_entry(hass, self)  # type: ignore
  File "/config/custom_components/reolink_dev/__init__.py", line 184, in async_unload_entry
    await unregister_webhook(hass, base.event_id)
  File "/config/custom_components/reolink_dev/__init__.py", line 176, in unregister_webhook
    _LOGGER.info("Unregistering webhook %s", info.name)
AttributeError: 'dict' object has no attribute 'name'
```

when adding/removing the integration to test.

**Comment by fwestenberg** [comment]
This should be fixed in v0.12. Please reopen if not.
